sky/jobs/controller.py

```diff
@@ -55,10 +57,16 @@ def __init__(self, job_id: int, dag_yaml: str,
         # TODO(zhwu): this assumes the specific backend.
         self._backend = cloud_vm_ray_backend.CloudVmRayBackend()
 
-        # pylint: disable=line-too-long
+        self._dag_graph = self._dag.get_graph()
+        self._task_queue = self._initialize_task_queue()
+        self._completed_tasks: Set[int] = set()
+        self._failed_tasks: Set[int] = set()
+        self._block_tasks: Set[int] = set()
```

Comment:
> Also, is it possible to use its name as the identifier? And related to previous PR, maybe instead of the related less meaningful timestamps, we could use its index in a topo order as its default name (if user does not specified one). e.g. `task_1`.

@cblmemo Could you clarify the second point? Not sure how using names relates to tasks here. Thanks!

Reply:
Currently we generate task name for tasks if user does not specify a name. The name is sth like `f'task_{current_timestamp}'` IIRC. Maybe we could change the naming convention to `f'task_{id_in_topo_order}'`

Reply:
@cblmemo Using topological order for task naming is problematic as DAG structure can change dynamically. When edges are added/removed, the topo order would change, causing task names to change. This is confusing and could break user code that references tasks by name generated. The current timestamp-based naming ensures stability and uniqueness.